zenml/utils/post_training/compare.py

```diff
@@ -31,16 +31,20 @@ class Application(param.Parameterized):
     slicing_metric_selector = param.ObjectSelector(default='', objects=[''])
     performance_metric_selector = param.ObjectSelector(objects=[])
 
-    def __init__(self, **params):
+    def __init__(self, datasource, **params):
         super(Application, self).__init__(**params)
 
         # lists
         result_list = []
         hparam_list = []
         repo: Repository = Repository.get_instance()
+        datasource = datasource
 
         # get all pipelines in this workspace
-        all_pipelines: List[TrainingPipeline] = repo.get_pipelines_by_type([
+        if datasource:
+            all_pipelines: List[TrainingPipeline] = repo.get_pipelines_by_datasource([datasource])
```

Comment:
If we do it this way, we might get other types of pipelines (like `Data`, `BatchInference` etc which are not comparable. We first need to filter the training pipelines and then only find the pipelines with the relevant datasource. We can do this via a simple for loop here after getting the pipelines. WDYT?